Week4/Solution.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Alyssa Lingad
+ * 
+ * MICP: Week 4
+ */
+
+import static org.junit.Assert.*;
+import org.junit.Test;
+import java.util.*;
+
+
+
+public class Solution
+{
+	public static boolean canBeSegmented(String str, HashSet d) throws Exception{
+		int i = 0;
+		if (str == null)
+			throw new Exception();
+		if (str.isEmpty())
+			return false;
+		if (d.isEmpty())
+			return false;
+		
+		int size = str.length();
+		String s = str.toLowerCase();
+		boolean check = false;
+		while (i < size){
+			int j = i + 2;
```

Comment:
I was having trouble with what to set j to. I set it to i+2 so that when I do substring(i,j) the substring would include the current char at i and the char at i+1. I did not think to test the case that there is a string inputted with 1 character. I fixed my code to account for that and tested for other cases. Thank you for the feedback!